sql_script/MS3_insertions.sql

```diff
@@ -1,30 +1,77 @@
+
 
 /*Add a new recipient to the recipient table 
--> NOTE: only allows insertion if postal code already exists in region*/
+	1) insert new_recipient postal_code into region table if it does not already exist
+		-> in this case, we would have to ask the user to provide municipality and province_or_territory
+	2) insert new recipient
+		-> only if recipient_id does not already exist in recipient table, otherwise, prompt warning. 
+			-> We could add option to replace recipient based on duplicate id if we have extra time*/
+INSERT INTO region
+VALUES (new_postal_code, new_municipality, new_province_or_territory)
+ON CONFLICT (postal_code) DO NOTHING;
+
 INSERT INTO recipient(id, street_address, postal_code)
 VALUES (new_id, new_street_address, new_postal_code);
 
 /*Add a new producer to the producer table
--> NOTE: only works when new_postal_code already exists in region table*/
-INSERT
-INTO producer (id, street_address, postal_code)
+	1) insert new_producer postal_code into region table if it does not already exist
+		-> in this case, we would have to ask the user to provide municipality and province_or_territory
+	2) insert new recipient
+		-> only if producer_id does not already exist in producer table, otherwise, prompt warning. 
+			-> We could add option to replace producer based on duplicate id if we have extra time*/
+
+INSERT INTO region
+VALUES (new_postal_code, new_municipality, new_province_or_territory)
+ON CONFLICT (postal_code) DO NOTHING;
+
+INSERT INTO producer (id, street_address, postal_code)
 VALUES (new_id, new_street_address, new_postal_code);
 
 /*Come up with values for a new container type, calculate its volume, insert that value into the dimensions table, then insert the container into container_type (has a FK to dimensions).
--> NOTE: only works if height, width, length tuple already exists in dimension table*/
+-> NOTE: warehouse referenced by new_warehouse_id must already exist. 
+	If not, give user the option to add the new warehouse before putting a container in it
+-> NOTE: occupied_volume is set to 0, as a new container will always be empty*/
+
+INSERT INTO dimensions
+VALUES (new_height, new_width, new_length, new_height * new_width * new_height)
+ON CONFLICT (height, width, length) DO NOTHING;
 
-INSERT
-INTO storage_container (id, warehouse_id, occupied_volume, height, width, length)
-VALUES (new_id, new_warehouse_id, new_occupied_volume, new_height, new_width, new_length);
+INSERT INTO storage_container (id, warehouse_id, occupied_volume, height, width, length)
+VALUES (new_id, new_warehouse_id, 0, new_height, new_width, new_length);
 
-/*Insert inventory, assign it to a container, update the container’s occupied volume, then update the warehouse’s total volume (??)
--> NOTE: only works if new_type_name already exists in type_name table*/
-INSERT
-INTO inventory (serial_num, container_id, type_name, weight, manufacture_date)
+/*UPDATE OCCUPIED VOLUME OF WAREHOUSE IN WHICH NEW CONTAINER IS BEING PLACED (?)
```

Comment:
This might cause problems and the syntax is a little complicated. We can possibly just leave out this query completely.

Reply:
Recap: Will leave insertions for last if we have time (along with GUI enhancements). Will leave implementation details for whoever decides to implement.